packages/mailchimp/test/gzip.test.js

```diff
@@ -0,0 +1,87 @@
+import { expect } from 'chai';
+import * as undici from 'undici'
+import {Readable } from 'node:stream';
+import tar from 'tar-stream';
+import fs from 'node:fs'
+import gunzip from 'gunzip-maybe';
+import fixtureData from './fixtures/data.json' assert { type: 'json' }
+
+describe('gzip', () => {
+  it('should extract a tarball as json', (cb) => {
+    const extract = tar.extract();
+    let data = '';
+
+    extract.on('entry', function(header, stream, next) {
+      // header.name // filename
+      stream.on('data', function(chunk) {
+        data += chunk
+      });
+      
+      stream.on('end', function() {
+        next();
+      });
+
+      stream.resume();
+    });
+
+    extract.on('finish', function() {
+      const json = JSON.parse(data);
+      expect(json).eql(fixtureData)
+      cb()
+    });
+    
+    
+    fs.createReadStream('./test/fixtures/data.tar').pipe(extract)
+
+  })
+
+  // this is the same as above, but with nicer syntax
+  it('should extract a tarball as json with async', async () => {
+    const extract = tar.extract();
+    let data = '';
+
+    fs.createReadStream('./test/fixtures/data.tar').pipe(extract)
+
+    for await (const entry of extract) {
+      console.log(entry.header.name) // filename
+      for await (const d of entry) {
+        data = d;
+      }
+      entry.resume();
+    }
+
+    const json = JSON.parse(data);
+    expect(json).eql(fixtureData)
+  });
+
+
+  // To get this test to pass, you have to run `npx serve` from the fixtures folder
+  it('should download and extract a tarball with undici fetch', async () => {
```

Comment:
This unit test relies on having an external server to server the `tgz` to undici. As the comment says, all you have to do is run `npx serve` from the fixtures folder.

But this isn't appropriate for CI.

We should be able to create a mock for this - we'd have to load the tarball as text, maybe set the appropriate headers, then return it to undici.

It's worth trying to get the mock working but if it takes more than an hour, we can keep and skip this test.